server/mqtt.go

```diff
@@ -3703,8 +4091,31 @@ func mqttDeliverMsgCbQos1(sub *subscription, pc *client, _ *Account, subject, re
 	}
 
 	topic := natsSubjectToMQTTTopic(strippedSubj)
+	retained := false // <>/<>
```

Comment:
Actually I have re-read the spec and I think that are some issues with the value of the RETAIN flag sent by server to client (at least on v3.1.1), but in that specific case, I believe it will have to be set to 0 (so false here would be correct). I need to spend a bit more time on that, create/update tests and fix the issue, but in a separate PR.